source/core.py

```diff
@@ -250,6 +250,48 @@ def onEndSession(evt):
 	import windowUtils
 	class MessageWindow(windowUtils.CustomWindow):
 		className = u"wxWindowClassNR"
+		batteryCache = None
+		REASON_INITIALIZE = 1
+		def __init__(self, windowName=None):
+			super(MessageWindow, self).__init__(windowName)
+			self.say_battery_status(reason=self.REASON_INITIALIZE)
+	
+		def windowProc(self, hwnd, msg, wParam, lParam):
+			import ui
+			#Just define these constants here, so we don't have to import win32con
+			WM_POWERBROADCAST = 0x218
+			WM_DISPLAYCHANGE = 0x7e
+			PBT_APMPOWERSTATUSCHANGE = 0xA

```

Comment:
Please define these at class level so they don't have to be redefined each time the function runs.